logging/Logger.cpp

```diff
@@ -0,0 +1,147 @@
+#include "Logger.hpp"
+#include "TimeFormatter.hpp"
+#include <iostream>
+#include <algorithm>
+
+namespace logging
+{
+    /**
+     * @brief Construct a new Logger:: Logger object
+     */
+    Logger::Logger()
+    {
+        this->_file_and_folder_handler.create_folder("logs");
+        std::string filename = TimeFormatter::get_time("YYYY-MM-DD-1.log");
+        for (int i = 1; this->_file_and_folder_handler.file_exist(filename); i++)
+        {
+            filename = TimeFormatter::get_time("YYYY-MM-DD-" + std::to_string(i) + ".log");
+        }
+        this->_file_and_folder_handler.create_file(filename);
+        this->_file_stream.open(this->_file_and_folder_handler.get_file_path(), std::ios::app);
+
+        this->_specification_level_in_file = {LogLevel::DEBUG, LogLevel::INFO, LogLevel::WARNING, LogLevel::ERROR, LogLevel::FATAL};
+        this->_specification_level_in_console = {LogLevel::DEBUG, LogLevel::INFO, LogLevel::WARNING, LogLevel::ERROR, LogLevel::FATAL};
+    }
+
+    /**
+     * @brief Destroy the Logger:: Logger object
+     *
+     * @note The file stream will be closed
+     */
+    Logger::~Logger()
+    {
+        this->_file_stream.close();
+    }
+
+    /**
+     * @brief Write a message in the log file
+     *
+     * @param message Message to write
+     *
+     * @note The message will be written in the log file with the current time
+     */
+    void Logger::log(LogLevel level, std::string message)
```

Comment:
Instead of having one function to log which takes a level in parameter, create one function per level, so that we can just call `logger.warn("A warning");` or `logger.error("An error");`, it would be much more readable